### PR TITLE
support passing cancelation_reason to .cancel_bookings

### PR DESCRIPTION
**Pull request by thilonel**

### CODE REVIEW

**Comment by thilonel** [comment]
I see that all cassettes are recorded against production, so we need to wait until the prod API supports this.

File: lib/bookingsync/api/client/bookings.rb [hunk 1]
```diff
@@ -70,9 +70,17 @@ def edit_booking(booking, options = {})
...
       # @return [NilClass] Returns nil on success.
-      def cancel_booking(booking, options = {})
-        delete "bookings/#{booking}"
+      # @example
+      #   @api.cancel_booking(booking_id)
+      #
+      # @example Providing cancelation_reason
+      #   @api.cancel_booking(booking_id, { cancelation_reason: "payment_failed" })
+      def cancel_booking(booking, options = nil)
+        options = { bookings: [options] } unless options.nil?
```
Comment (Azdaroth) [inline lib/bookingsync/api/client/bookings.rb]: I guess `options = { bookings: [options] } if options?` would work as well and be cleaner.

File: spec/fixtures/cassettes/BookingSync_API_Client_Bookings/_cancel_booking/cancels_given_booking_and_passes_options_along_as_bookings_payload.yml [hunk 2]
```diff
@@ -68,6 +68,6 @@ http_interactions:
     body:
       encoding: UTF-8
       string: ''
-    http_version: 
+    http_version:
   recorded_at: Mon, 12 Dec 2016 12:13:17 GMT
```
Comment (Azdaroth) [inline spec/fixtures/cassettes/BookingSync_API_Client_Bookings/_cancel_booking/cancels_given_booking_and_passes_options_along_as_bookings_payload.yml]: I see what you did here :doge: This cassette must be re-recorded.

Comment (thilonel) [inline spec/fixtures/cassettes/BookingSync_API_Client_Bookings/_cancel_booking/cancels_given_booking_and_passes_options_along_as_bookings_payload.yml]: Let's make a separate issue for that, cause it's not that easy. I tried...

Comment (Azdaroth) [inline spec/fixtures/cassettes/BookingSync_API_Client_Bookings/_cancel_booking/cancels_given_booking_and_passes_options_along_as_bookings_payload.yml]: I would rather record it myself than let it be merged like that :P  Shoudl I handle it then?

**Pull request merged**